packages/remix-react/components.tsx

```diff
@@ -778,26 +783,10 @@ export function Scripts(props: ScriptProps) {
 
     let routeModulesScript = !isStatic
       ? " "
-      : `${
-          manifest.hmr?.runtime
-            ? `import ${JSON.stringify(manifest.hmr.runtime)};`
-            : ""
-        }import ${JSON.stringify(manifest.url)};
-${matches
-  .map(
-    (match, index) =>
-      `import * as route${index} from ${JSON.stringify(
-        manifest.routes[match.route.id].module
-      )};`
-  )
-  .join("\n")}
-window.__remixRouteModules = {${matches
-          .map(
-            (match, index) => `${JSON.stringify(match.route.id)}:route${index}`
-          )
-          .join(",")}};
-
-import(${JSON.stringify(manifest.entry.module)});`;
+      : remixEntryJsString(
+          manifest,
+          matches.map((m) => m.route.id)
+        );
```

Comment:
Internal `<Scripts>` logic extracted to internal `UNSAFE` helpers in `@remix-run/server-runtime/serverHandoff` so we can also leverage them from the vite plugin